app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3557,14 +3557,19 @@ private void buildCaptionMenu(@NonNull final List<String> availableLanguages) {
         }
 
         // apply caption language from previous user preference
-        final List<String> selectedPreferredLanguages =
-                trackSelector.getParameters().preferredTextLanguages;
         final String userPreferredLanguage =
                 prefs.getString(context.getString(R.string.caption_user_set_key), null);
         final int textRendererIndex = getCaptionRendererIndex();
 
-        if (userPreferredLanguage != null
-                && !selectedPreferredLanguages.contains(userPreferredLanguage)
+        if (userPreferredLanguage == null) {
```

Comment:
1. Please add some comments why this stuff is happening
2. What is happening if ``textRendererIndex == RENDERER_UNAVAILABLE``?

Reply:
1. Fixed.
2. Fixed. Though nothing would happen in practice since here `RENDERER_UNAVAILABLE` is being disabled. But it is bad form and inconsistent with the other checks.